### PR TITLE
feat: integrate phone number challenge api

### PR DESCRIPTION
**Pull request by utkarsha-deriv**
## Changes:

- create hook for requesting OTP on whatsapp or sms: `useGetPhoneNumberOTP`
- integrate api using this hook
- add types for phone_number_challenge api
- add constants for verification services: whatsapp, sms
- add testcases for the `useGetPhoneNumberOTP` hook

### CODE REVIEW

File: packages/account/src/Sections/Profile/PhoneVerification/confirm-phone-number.tsx [hunk 1]
```diff
@@ -4,24 +4,27 @@ import { Button, Text } from '@deriv-com/quill-ui';
 import { Localize, localize } from '@deriv/translations';
 import { Input } from '@deriv/components';
 import { observer, useStore } from '@deriv/stores';
+import { useGetPhoneNumberOTP } from '@deriv/hooks';
 
 const ConfirmPhoneNumber = observer(() => {
+    const { requestOnSMS, requestOnWhatsApp, ...rest } = useGetPhoneNumberOTP();
```
Comment (yauheni-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/confirm-phone-number.tsx:10]: do we need any 'loading' functionality? block the ui while loading, disable buttons etc...

Comment (utkarsha-deriv) [reply packages/account/src/Sections/Profile/PhoneVerification/confirm-phone-number.tsx:10]: yes, i think we will cover that in other cards 🤔 @amina-deriv ?

**Review by yauheni-deriv — APPROVED** [review]

File: packages/account/src/Sections/Profile/PhoneVerification/confirm-phone-number.tsx [hunk 2]
```diff
@@ -4,24 +4,27 @@ import { Button, Text } from '@deriv-com/quill-ui';
...
     return (
         <PhoneVerificationCard>
             <Text bold>
                 <Localize i18n_default_text='Confirm your phone number' />
             </Text>
-            <Input label={localize('Phone number')} value={phoneNumber} />
+            <Input label={localize('Phone number')} value={phone_number} />
             <div className='phone-verification__card--buttons_container'>
-                <Button variant='secondary' color='black' fullWidth size='lg'>
+                <Button variant='secondary' color='black' fullWidth size='lg' onClick={requestOnSMS}>
```
Comment (suisin-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/confirm-phone-number.tsx:22]: Can we add test case for this API calls 🙏

File: packages/hooks/src/useGetPhoneNumberOTP.ts
Comment (likhith-deriv) [inline packages/hooks/src/useGetPhoneNumberOTP.ts:1]: 👍

**Review by likhith-deriv — APPROVED** [review]

File: packages/hooks/src/useGetPhoneNumberOTP.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    return {
+        data: data?.phone_number_challenge,
+        requestOnWhatsApp,
+        requestOnSMS,
+        ...rest,
+    };
+};
+
+export default useGetPhoneNumberOTP;
```
Comment (amina-deriv) [inline packages/hooks/src/useGetPhoneNumberOTP.ts:23]: Please add test cases for the hook

File: packages/account/src/Sections/Profile/PhoneVerification/confirm-phone-number.tsx (hunk 2 above)
Comment (suisin-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/confirm-phone-number.tsx:22]: Can we mock the call in `confirm-phone-number.spec.tsx` 🙏

**Review by amina-deriv — APPROVED** [review]

**Review by suisin-deriv — APPROVED** [review]

**Pull request merged**